From 4e825165d3de4a21debffcdce8867fbab877d676 Mon Sep 17 00:00:00 2001
From: Andi Kleen <andi@firstfloor.org>
Date: Tue, 13 Oct 2020 15:54:56 -0700
Subject: x86/livepatch/lto: Disable live patching with LTO

Not supported by gcc 10 so far, causes compiler "sorry" messages.

Other than the compiler support there shouldn't be any barriers for live
patching LTOed kernels, although it might be more difficult to create
patches for larger functions.

Signed-off-by: Andi Kleen <ak@linux.intel.com>
---
 kernel/livepatch/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/livepatch/Kconfig b/kernel/livepatch/Kconfig
index 53d51ed619a3d..8c004f211a442 100644
--- a/kernel/livepatch/Kconfig
+++ b/kernel/livepatch/Kconfig
@@ -12,6 +12,7 @@ config LIVEPATCH
 	depends on KALLSYMS_ALL
 	depends on HAVE_LIVEPATCH
 	depends on !TRIM_UNUSED_KSYMS
+	depends on !LTO_GCC # not supported in gcc 10
 	help
 	  Say Y here if you want to support kernel live patching.
 	  This option has no runtime impact until a kernel "patch"
-- 
cgit 1.2.3-1.el7

